features/lorawan/LoRaWANStack.cpp

```diff
@@ -353,7 +353,14 @@ lora_mac_status_t LoRaWANStack::send_frame_to_mac()
         mcps_req.f_buffer = _tx_msg.f_buffer;
         mcps_req.f_buffer_size = _tx_msg.f_buffer_size;
         mcps_req.req.confirmed.nb_trials = _tx_msg.message_u.confirmed.nb_trials;
-        mcps_req.req.confirmed.datarate = default_datarate.Value;
+
+        mib_get_params.type = LORA_MIB_CHANNELS_DATARATE;
+           if(mib_get_request(&mib_get_params) != LORA_MAC_STATUS_OK) {
```

Comment:
Indentation not right for this block.

Reply:
Okay I will make sure this gets rectified in next release. I have taken a note.